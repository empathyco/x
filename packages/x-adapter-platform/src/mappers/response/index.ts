export * from './search-response.mapper';
