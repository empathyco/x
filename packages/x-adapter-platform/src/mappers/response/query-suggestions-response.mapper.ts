import { schemaMapperFactory } from '@empathyco/x-adapter-next';
import { QuerySuggestionsResponse } from '@empathyco/x-types';
// eslint-disable-next-line max-len
import { querySuggestionsResponseMutableSchema } from '../../schemas/response/query-suggestions-response.schema';
// eslint-disable-next-line max-len
import { PlatformQuerySuggestionsResponse } from '../../types/response/query-suggestions-response.model';

export const querySuggestionsResponseMapper = schemaMapperFactory<
  PlatformQuerySuggestionsResponse,
  QuerySuggestionsResponse
>(querySuggestionsResponseMutableSchema);
