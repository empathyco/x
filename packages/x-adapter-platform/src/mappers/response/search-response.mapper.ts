import { createMutableSchema, Schema, schemaMapperFactory } from '@empathyco/x-adapter-next';
import { PlatformSearchResponse, SearchResponse } from '../../types';

const searchResponseSchema: Schema<PlatformSearchResponse, SearchResponse> = {
  results: 'catalog.content',
  facets: 'catalog.facets',
  totalResults: 'catalog.numFound',
  spellcheck: 'catalog.spellchecked',
  banners: 'banner.content',
  promoteds: 'promoted.content',
  redirections: 'direct.content',
  queryTagging: 'catalog.tagging.query'
};

export const mutableSearchResponseSchema = createMutableSchema(searchResponseSchema);

export const searchResponseMapper = schemaMapperFactory<PlatformSearchResponse, any>(
  mutableSearchResponseSchema
);
