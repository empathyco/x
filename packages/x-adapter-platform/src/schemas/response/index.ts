export * from './query-suggestions-response.schema';
