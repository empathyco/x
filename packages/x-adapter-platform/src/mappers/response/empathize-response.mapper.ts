import { createMutableSchema, Schema, schemaMapperFactory } from '@empathyco/x-adapter-next';
import { EmpathizeResponse, PlatformEmpathizeResponse } from '../../types';
import { suggestionMutableSchema } from '../../schemas';

export const empathizeResponseSchema: Schema<PlatformEmpathizeResponse, EmpathizeResponse> = {
  suggestions: {
    $path: 'topTrends.content',
    $subSchema: suggestionMutableSchema
  },
  spellcheck: 'topTrends.spellcheck'
};

export const empathizeResponseMutableSchema = createMutableSchema(empathizeResponseSchema);

export const empathizeResponseMapper = schemaMapperFactory<
  PlatformEmpathizeResponse,
  EmpathizeResponse
>(empathizeResponseMutableSchema);
