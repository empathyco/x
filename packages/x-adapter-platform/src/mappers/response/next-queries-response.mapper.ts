import { schemaMapperFactory } from '@empathyco/x-adapter-next';
import { NextQueriesResponse } from '@empathyco/x-types';
import { nextQueriesResponseMutableSchema } from '../../schemas/response';
import { PlatformNextQueriesResponse } from '../../types/response/next-queries-response.model';

export const nextQueriesResponseMapper = schemaMapperFactory<
  PlatformNextQueriesResponse,
  NextQueriesResponse
>(nextQueriesResponseMutableSchema);
