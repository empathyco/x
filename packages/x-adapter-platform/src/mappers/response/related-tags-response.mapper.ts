import { createMutableSchema, Schema, schemaMapperFactory } from '@empathyco/x-adapter-next';
import { PlatformRelatedTagsResponse, RelatedTagsResponse } from '../../types/response.types';
import { relatedTagMutableSchema } from '../../schemas/related-tag.schema';

export const relatedTagsResponseSchema: Schema<PlatformRelatedTagsResponse, RelatedTagsResponse> = {
  relatedTags: {
    $path: 'data.relatedtags',
    $subSchema: relatedTagMutableSchema
  }
};

export const relatedTagsResponseMutableSchema = createMutableSchema(relatedTagsResponseSchema);

export const relatedTagsResponseMapper = schemaMapperFactory<
  PlatformRelatedTagsResponse,
  RelatedTagsResponse
>(relatedTagsResponseMutableSchema);
