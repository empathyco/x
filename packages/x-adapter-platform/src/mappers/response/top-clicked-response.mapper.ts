import { createMutableSchema, Schema, schemaMapperFactory } from '@empathyco/x-adapter-next';
import { PlatformTopClickedResponse, TopClickedResponse } from '../../types/response.types';
import { resultMutableSchema } from '../../schemas/result.schema';

export const topClickedResponseSchema: Schema<PlatformTopClickedResponse, TopClickedResponse> = {
  results: {
    $path: 'topclicked.content',
    $subSchema: resultMutableSchema
  }
};

export const topClickedResponseMutableSchema = createMutableSchema(topClickedResponseSchema);

export const topClickedResponseMapper = schemaMapperFactory<
  PlatformTopClickedResponse,
  TopClickedResponse
>(topClickedResponseMutableSchema);
