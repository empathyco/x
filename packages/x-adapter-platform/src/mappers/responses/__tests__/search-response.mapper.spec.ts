import { searchResponseMapper } from '../search-response.mapper';
import { platformSearchResponse } from '../../../__tests__/__fixtures__/search.response';

describe('searchResponseMapper tests', () => {
  it('should map the response', () => {
    expect(searchResponseMapper(platformSearchResponse, {})).toMatchSnapshot();
  });
});
