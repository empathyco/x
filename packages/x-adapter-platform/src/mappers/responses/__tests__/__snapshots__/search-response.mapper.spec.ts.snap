// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchResponseMapper tests should map the response 1`] = `
Object {
  "banners": Array [
    Object {
      "id": "61f9223221c06701c2f98c17",
      "image": "https://assets.empathy.co/images-demo/31335.jpg",
      "modelName": "Banner",
      "tagging": Object {
        "click": Object {
          "params": Object {
            "filtered": "false",
            "follow": false,
            "lang": "en",
            "origin": "url:external",
            "page": "1",
            "position": "1",
            "productId": "61f9223221c06701c2f98c17",
            "q": "jeans",
            "scope": "desktop",
            "spellcheck": "false",
            "title": "banner title",
            "url": "https://assets.empathy.co/",
          },
          "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/click",
        },
        "query": Object {
          "params": Object {
            "filtered": "false",
            "follow": false,
            "lang": "en",
            "origin": "url:external",
            "page": "1",
            "position": "1",
            "productId": "61f9223221c06701c2f98c17",
            "q": "jeans",
            "scope": "desktop",
            "spellcheck": "false",
            "title": "banner title",
            "url": "https://assets.empathy.co/",
          },
          "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/query",
        },
      },
      "title": "Banner title",
      "url": "https://assets.empathy.co/",
    },
  ],
  "facets": Array [
    Object {
      "filters": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "facetId": "categoryPaths",
                      "id": "categoryIds:78d9b7366__8a4e61a33_aa_bb",
                      "label": "Added 2",
                      "modelName": "HierarchicalFilter",
                      "parentId": "categoryIds:78d9b7366__8a4e61a33_aa",
                      "selected": false,
                      "totalResults": 1,
                    },
                  ],
                  "facetId": "categoryPaths",
                  "id": "categoryIds:78d9b7366__8a4e61a33_aa",
                  "label": "Added",
                  "modelName": "HierarchicalFilter",
                  "parentId": "categoryIds:78d9b7366__8a4e61a33",
                  "selected": false,
                  "totalResults": 1,
                },
              ],
              "facetId": "categoryPaths",
              "id": "categoryIds:78d9b7366__8a4e61a33",
              "label": "Bottomwear",
              "modelName": "HierarchicalFilter",
              "parentId": "categoryIds:78d9b7366",
              "selected": false,
              "totalResults": 637,
            },
            Object {
              "facetId": "categoryPaths",
              "id": "categoryIds:78d9b7366__e2f94a4ea",
              "label": "Topwear",
              "modelName": "HierarchicalFilter",
              "parentId": "categoryIds:78d9b7366",
              "selected": false,
              "totalResults": 99,
            },
          ],
          "facetId": "categoryPaths",
          "id": "categoryIds:78d9b7366",
          "label": "Apparel",
          "modelName": "HierarchicalFilter",
          "parentId": null,
          "selected": false,
          "totalResults": 736,
        },
        Object {
          "facetId": "categoryPaths",
          "id": "categoryIds:b08648dbd",
          "label": "Accessories",
          "modelName": "HierarchicalFilter",
          "parentId": null,
          "selected": false,
          "totalResults": 43,
        },
        Object {
          "children": Array [
            Object {
              "facetId": "categoryPaths",
              "id": "categoryIds:ffc61e1e9_aa",
              "label": "Added",
              "modelName": "HierarchicalFilter",
              "parentId": "categoryIds:ffc61e1e9",
              "selected": false,
              "totalResults": 1,
            },
          ],
          "facetId": "categoryPaths",
          "id": "categoryIds:ffc61e1e9",
          "label": "Personal Care",
          "modelName": "HierarchicalFilter",
          "parentId": null,
          "selected": false,
          "totalResults": 9,
        },
        Object {
          "facetId": "categoryPaths",
          "id": "categoryIds:e5eef62d8",
          "label": "Footwear",
          "modelName": "HierarchicalFilter",
          "parentId": null,
          "selected": false,
          "totalResults": 6,
        },
      ],
      "id": "categoryPaths",
      "label": "categoryPaths",
      "modelName": "HierarchicalFacet",
    },
    Object {
      "filters": Array [
        Object {
          "facetId": "price",
          "id": "price:10.0-20.0",
          "label": "10.0-20.0",
          "modelName": "NumberRangeFilter",
          "range": Object {
            "max": 20,
            "min": 10,
          },
          "selected": false,
          "totalResults": 97,
        },
        Object {
          "facetId": "price",
          "id": "price:20.0-30.0",
          "label": "20.0-30.0",
          "modelName": "NumberRangeFilter",
          "range": Object {
            "max": 30,
            "min": 20,
          },
          "selected": false,
          "totalResults": 80,
        },
        Object {
          "facetId": "price",
          "id": "price:30.0-40.0",
          "label": "30.0-40.0",
          "modelName": "NumberRangeFilter",
          "range": Object {
            "max": 40,
            "min": 30,
          },
          "selected": false,
          "totalResults": 85,
        },
        Object {
          "facetId": "price",
          "id": "price:40.0-50.0",
          "label": "40.0-50.0",
          "modelName": "NumberRangeFilter",
          "range": Object {
            "max": 50,
            "min": 40,
          },
          "selected": false,
          "totalResults": 75,
        },
        Object {
          "facetId": "price",
          "id": "price:50.0-60.0",
          "label": "50.0-60.0",
          "modelName": "NumberRangeFilter",
          "range": Object {
            "max": 60,
            "min": 50,
          },
          "selected": false,
          "totalResults": 88,
        },
        Object {
          "facetId": "price",
          "id": "price:60.0-70.0",
          "label": "60.0-70.0",
          "modelName": "NumberRangeFilter",
          "range": Object {
            "max": 70,
            "min": 60,
          },
          "selected": false,
          "totalResults": 62,
        },
        Object {
          "facetId": "price",
          "id": "price:70.0-80.0",
          "label": "70.0-80.0",
          "modelName": "NumberRangeFilter",
          "range": Object {
            "max": 80,
            "min": 70,
          },
          "selected": false,
          "totalResults": 84,
        },
        Object {
          "facetId": "price",
          "id": "price:80.0-90.0",
          "label": "80.0-90.0",
          "modelName": "NumberRangeFilter",
          "range": Object {
            "max": 90,
            "min": 80,
          },
          "selected": false,
          "totalResults": 86,
        },
        Object {
          "facetId": "price",
          "id": "price:90.0-100.0",
          "label": "90.0-100.0",
          "modelName": "NumberRangeFilter",
          "range": Object {
            "max": 100,
            "min": 90,
          },
          "selected": false,
          "totalResults": 79,
        },
      ],
      "id": "price",
      "label": "price",
      "modelName": "NumberRangeFacet",
    },
    Object {
      "filters": Array [
        Object {
          "facetId": "gender",
          "id": "gender:men",
          "label": "men",
          "modelName": "SimpleFilter",
          "selected": false,
          "totalResults": 421,
        },
        Object {
          "facetId": "gender",
          "id": "gender:women",
          "label": "women",
          "modelName": "SimpleFilter",
          "selected": false,
          "totalResults": 247,
        },
        Object {
          "facetId": "gender",
          "id": "gender:boys",
          "label": "boys",
          "modelName": "SimpleFilter",
          "selected": false,
          "totalResults": 35,
        },
        Object {
          "facetId": "gender",
          "id": "gender:girls",
          "label": "girls",
          "modelName": "SimpleFilter",
          "selected": false,
          "totalResults": 28,
        },
        Object {
          "facetId": "gender",
          "id": "gender:unisex",
          "label": "unisex",
          "modelName": "SimpleFilter",
          "selected": false,
          "totalResults": 5,
        },
      ],
      "id": "gender",
      "label": "gender",
      "modelName": "SimpleFacet",
    },
  ],
  "partialResults": Array [
    Object {
      "query": "shirt",
      "results": Array [
        Object {
          "id": "31335-U",
          "identifier": Object {},
          "isWishlisted": false,
          "modelName": "Result",
          "name": "Locomotive Men Washed Blue Jeans",
          "price": Object {
            "futureValue": 10,
            "hasDiscount": false,
            "originalValue": 10,
            "value": 10,
          },
          "rating": Object {
            "value": null,
          },
          "tagging": Object {
            "add2cart": Object {
              "params": Object {
                "filtered": "false",
                "follow": false,
                "lang": "en",
                "origin": "url:external",
                "page": "1",
                "position": "1",
                "productId": "31335-U",
                "q": "jeans",
                "scope": "desktop",
                "spellcheck": "false",
                "title": "Locomotive Men Washed Blue Jeans",
                "url": "https://assets.empathy.co/images-demo/31335.jpg",
              },
              "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/add2cart",
            },
            "checkout": Object {
              "params": Object {
                "filtered": "false",
                "follow": false,
                "lang": "en",
                "origin": "url:external",
                "page": "1",
                "position": "1",
                "productId": "31335-U",
                "q": "jeans",
                "scope": "desktop",
                "spellcheck": "false",
                "title": "Locomotive Men Washed Blue Jeans",
                "url": "https://assets.empathy.co/images-demo/31335.jpg",
              },
              "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/checkout",
            },
            "click": Object {
              "params": Object {
                "filtered": "false",
                "follow": false,
                "lang": "en",
                "origin": "url:external",
                "page": "1",
                "position": "1",
                "productId": "31335-U",
                "q": "jeans",
                "scope": "desktop",
                "spellcheck": "false",
                "title": "Locomotive Men Washed Blue Jeans",
                "url": "https://assets.empathy.co/images-demo/31335.jpg",
              },
              "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/click",
            },
            "displayClick": Object {
              "params": Object {
                "displayId": "jeans",
                "filtered": "false",
                "follow": false,
                "lang": "en",
                "origin": "url:external",
                "page": "1",
                "position": "1",
                "productId": "31335-U",
                "scope": "desktop",
                "spellcheck": "false",
                "title": "Locomotive Men Washed Blue Jeans",
                "url": "https://assets.empathy.co/images-demo/31335.jpg",
              },
              "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/displayClick",
            },
          },
          "type": "Default",
          "url": "https://assets.empathy.co/images-demo/31335.jpg",
        },
      ],
      "totalResults": 84,
    },
  ],
  "promoteds": Array [
    Object {
      "id": "61f9223221c06701c2f98c17",
      "image": "https://assets.empathy.co/images-demo/31335.jpg",
      "modelName": "Promoted",
      "tagging": Object {
        "query": Object {
          "params": Object {
            "follow": false,
          },
          "url": "",
        },
      },
      "title": "Promoted title",
      "url": "https://assets.empathy.co/",
    },
  ],
  "queryTagging": Object {
    "params": Object {
      "filtered": "true",
      "follow": false,
      "lang": "en",
      "origin": "url:external",
      "page": "1",
      "q": "jeans",
      "scope": "desktop",
      "spellcheck": "false",
      "totalHits": "686",
    },
    "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/query",
  },
  "redirections": Array [
    Object {
      "id": "bbf9223221c06701c2f98c1a",
      "modelName": "Redirection",
      "tagging": Object {
        "click": Object {
          "params": Object {
            "follow": false,
          },
          "url": "",
        },
      },
      "url": "https://api.staging.empathy.co",
    },
  ],
  "results": Array [
    Object {
      "id": "31335-U",
      "identifier": Object {
        "value": "31335-U",
      },
      "isWishlisted": false,
      "modelName": "Result",
      "name": "Locomotive Men Washed Blue Jeans",
      "price": Object {
        "futureValue": 10,
        "hasDiscount": false,
        "originalValue": 10,
        "value": 10,
      },
      "rating": Object {
        "value": null,
      },
      "tagging": Object {
        "add2cart": Object {
          "params": Object {
            "filtered": "false",
            "follow": false,
            "lang": "en",
            "origin": "url:external",
            "page": "1",
            "position": "1",
            "productId": "31335-U",
            "q": "jeans",
            "scope": "desktop",
            "spellcheck": "false",
            "title": "Locomotive Men Washed Blue Jeans",
            "url": "https://assets.empathy.co/images-demo/31335.jpg",
          },
          "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/add2cart",
        },
        "checkout": Object {
          "params": Object {
            "filtered": "false",
            "follow": false,
            "lang": "en",
            "origin": "url:external",
            "page": "1",
            "position": "1",
            "productId": "31335-U",
            "q": "jeans",
            "scope": "desktop",
            "spellcheck": "false",
            "title": "Locomotive Men Washed Blue Jeans",
            "url": "https://assets.empathy.co/images-demo/31335.jpg",
          },
          "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/checkout",
        },
        "click": Object {
          "params": Object {
            "filtered": "false",
            "follow": false,
            "lang": "en",
            "origin": "url:external",
            "page": "1",
            "position": "1",
            "productId": "31335-U",
            "q": "jeans",
            "scope": "desktop",
            "spellcheck": "false",
            "title": "Locomotive Men Washed Blue Jeans",
            "url": "https://assets.empathy.co/images-demo/31335.jpg",
          },
          "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/click",
        },
        "displayClick": Object {
          "params": Object {
            "displayId": "jeans",
            "filtered": "false",
            "follow": false,
            "lang": "en",
            "origin": "url:external",
            "page": "1",
            "position": "1",
            "productId": "31335-U",
            "scope": "desktop",
            "spellcheck": "false",
            "title": "Locomotive Men Washed Blue Jeans",
            "url": "https://assets.empathy.co/images-demo/31335.jpg",
          },
          "url": "https://api.staging.empathy.co/tagging/v1/track/empathy/displayClick",
        },
      },
      "type": "Default",
      "url": "https://assets.empathy.co/images-demo/31335.jpg",
    },
  ],
  "spellcheck": "",
  "totalResults": 1,
}
`;
